src/main/scala/lasp/hapi/util/Time.scala

```diff
@@ -0,0 +1,54 @@
+package lasp.hapi.util
+
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+import java.time.format.DateTimeFormatterBuilder
+import java.time.format.DateTimeParseException
+import java.time.temporal.ChronoField.MONTH_OF_YEAR
+import java.time.temporal.ChronoField.DAY_OF_MONTH
+import java.time.temporal.ChronoField.HOUR_OF_DAY
+import java.time.temporal.ChronoField.MINUTE_OF_HOUR
+import java.time.temporal.ChronoField.SECOND_OF_MINUTE
+
+import cats.implicits._
+
+object Time {
+
+  /**
+   * Parses a HAPI time string.
+   *
+   * @param str HAPI time string
+   */
+  def parse(str: String): Option[LocalDateTime] =
+    tryParse(str, format) <+> tryParse(str, formatOrd)
```

Comment:
See [`SemigroupK`](https://typelevel.org/cats/typeclasses/semigroupk.html).

This will return the first `Some`, which is the first successful parse.